src/settings/appsettings.cpp

```diff
@@ -664,6 +664,51 @@ QOnlineTts::Emotion AppSettings::defaultEmotion(QOnlineTranslator::Engine engine
     }
 }
 
+QMap<QOnlineTranslator::Language, QLocale::Country> AppSettings::regions(QOnlineTranslator::Engine engine) const
+{
+    switch (engine) {
+    case QOnlineTranslator::Google: {
+        QMap<QOnlineTranslator::Language, QLocale::Country> regions;
+        for (int i = 1; i <= QOnlineTranslator::Zulu; ++i)
+            regions[static_cast<QOnlineTranslator::Language>(i)] = m_settings->value("Translation/GoogleRegion/" + QOnlineTranslator::languageName(static_cast<QOnlineTranslator::Language>(i)), QLocale::AnyCountry).value<QLocale::Country>();
+        return regions;
```

Comment:
Maybe just set/get `QMap` to/from `m_settings` directly?

Reply:
I just tried this and got the following error:
error: no viable conversion from 'const QMap<QOnlineTranslator::Language, QLocale::Country>' to 'const QVariant'
Probably `QSettings` does not support this? I use Qt 5.15.2

Reply:
Right, it looks like it supports only [QMap<QString, QVariant>](https://doc.qt.io/qt-5/qvariant.html#QVariant-22)...
Then let's leave it as is, but please iterate only over valid regions and use built-in string formatting to represent it more nicely.